### DISCUSSION TITLE
logic for volume based routing

### DISCUSSION THREAD

**Discussion by euluizaraujo** [juspay/hyperswitch#3040] [Q&A] [answered]
Hello, could you explain the distribution logic of volume-based routing?  How does the algorithm logic for percentage distribution work?

**Comment by vspecky** [comment, marked as answer]
@euluizaraujo For volume-based routing we sample from a weighted distribution (where the weights are the percentage volume for each connector) to decide the connector. So basically if your distribution is `70% Stripe, 30% Adyen`, the algorithm has a `70%` chance of picking Stripe, and a `30%` change of picking Adyen when invoked. i.e. the selection is ad-hoc and not dependent on any previous selection. As a result, you will most probably observe some skew in the connector choice of your transactions with a small volume of transactions, but the skew will eventually resolve itself as the number of your transactions grows under a particular volume-based routing scheme.

**Reply by euluizaraujo** [reply]
@vspecky Perfect, I saw that they are developing the cost-based routing option. Will this option automatically collect the cost via API or will the cost have to be entered by the retailer when activating and configuring the connector?

**Reply by vspecky** [reply]
@euluizaraujo The gist of it is that we'll maintain any default pricing scheme of a connector (if available) on our end which can be used as your own configuration (one-click setup). Additionally you would also have the ability to configure any custom pricing scheme you have with the connector. Is this a feature that you have a requirement for?

**Reply by euluizaraujo** [reply]
@vspecky Perfect, if you want to make the figma file of the payout management functionality available in the control center, I can help implement it.  Are you planning to release this functionality this month?

**Reply by vspecky** [reply]
@euluizaraujo Hey, If you want to discuss payouts, I'd suggest creating a separate thread. As far as the design of the cost based routing feature goes, we are already in the process of designing it, but we're open to suggestions as well!
